atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresDdlTable.java

```diff
@@ -118,10 +124,56 @@ public void checkDatabaseVersion() {
 
     @Override
     public void compactInternally() {
+        if (config.compactIntervalMillis() > 0 && checkIfTableHasNotBeenCompactedForCompactIntervalMillis()) {
+            runCompactOnTable();
```

Comment:
Have a Semaphore that guards this method call (make sure to release the permit in a finally block)

Reply:
Semaphore would work, though probably just a `Lock` would be sufficient.

Also @gmaretic I'm guessing you want to do `tryLock()` / `tryAcquire()` rather than the blocking versions?